src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -867,6 +873,30 @@ public bool StartAsyncOperation(SocketAsyncContext context, TOperation operation
                         return false;
                     }
                 }
+
+                static void HandleFailedRegistration(SocketAsyncContext context, TOperation operation, Interop.Error error)
+                {
+                    Debug.Assert(error != Interop.Error.SUCCESS);
+
+                    // macOS: kevent returns EPIPE when adding pipe fd for which the other end is closed.
+                    if (error == Interop.Error.EPIPE)
+                    {
+                        // Retry the operation.
```

Comment:
Can you elaborate in the comment as to why we want to retry here? Is the idea that between the last time we tried and now, the other end may have closed, and so now when we try again, we should be able to successfully complete the operation?  And then if we can't for some reason we fall through to throwing because something definitely went wrong?

Reply:
Yes. I'll add this in the comments.